yoke/assets/joypad/base.js

```diff
@@ -722,22 +734,31 @@ function Joypad() {
     if (this.controls.deviceOrientation.length > 0) {
         window.addEventListener('deviceorientation', Motion.prototype.onDeviceOrientation, true);
     }
-    this.controls.byNumID.forEach(function(c) {
-        var id = c.element.id;
-        if (id[0] == 'b' || id[0] == 'a') {
-            c.neighbors = findNeighbors(this.gridAreas, c.element.id)
-                .filter(function(x) { return (x[0] == 'b' || x[0] == 'a'); });
-            c.neighbors = c.neighbors.map(function(x) {
-                return this.controls.byMnemonicID[x];
-            }, this);
+    // Users may press two buttons at the same time, or slide the finger towards neighboring buttons.
+    // Overlapping buttons will respond to this, but faraway buttons will not.
+    // This limit may change in the future.
+    this.controls.byNumID.forEach(function(c, _, arr) {
+        if (c.element.id[0] == 'b' || c.element.id[0] == 'a') {
+            c.neighbors = arr.filter(function(p) {
+                if (p.element.id[0] != 'b' && p.element.id[0] != 'a') {
+                    return false;
+                }
+                if (c.offset.x > p.offset.xMax ||
+                    c.offset.xMax < p.offset.x ||
+                    c.offset.y > p.offset.yMax ||
+                    c.offset.yMax < p.offset.y) {
+                    return false;
+                }
+                return true;
+            });
         }
     }, this);
     // Announce current controls:
     var kernelEvents = this.controls.byNumID.map(function(c) { return c.element.id; }).join(',');
     if (this.debugLabel != null) {
-        this.debugLabel.element.innerHTML = kernelEvents;
+        this.debugLabel.element.innerHTML = kernelEvents + '<br/>' + this.debugMessage;
```

Comment:
I'm not sure if it's a good idea, but we could [add `white-space: pre-wrap / pre-line;` style](https://css-tricks.com/almanac/properties/w/whitespace/) to debug label, and then just use `\n` instead of `<br/>`.

Reply:
It looks awesome to me. It's simpler to program, and makes more sense for this kind of messages.

Even if it looked odd, debug messages aren't really intended to be used every time anyways. You'll have noticed I even removed the debug ticker at `gamepad.css`

Bundled in commit 537187abcb68a2874b3fa65512a77634b964a2a4 too.